api_types.ts

```diff
@@ -805,6 +807,65 @@ export interface Logger {
   error(message: string, ...args: LoggerParamType[]): void;
 }
 
+export enum InvocationErrorType {
+  Timeout = 'Timeout',
+
+  ResponseTooLarge = 'ResponseTooLarge',
+
+  HttpStatusError = 'HttpStatusError',
+
+  /**
+   * Could mean 3rd party API rate limit or a rate limit imposed by Coda.
+   */
+  RateLimitExceeded = 'RateLimitExceeded',
+
+  Unknown = 'Unknown',
+}
+
+interface BaseInvocationError {
+  type: InvocationErrorType;
+
+  message?: string;
+
+  errorData?: string | object;
+}
+
+export type HttpStatusInvocationError = BaseInvocationError & {
+  type: InvocationErrorType.HttpStatusError;
+  statusCode: HttpStatusCode;
+};
+
+export type RateLimitExceededInvocationError = BaseInvocationError & {
+  type: InvocationErrorType.RateLimitExceeded;
+};
+
+export type TimeoutInvocationError = BaseInvocationError & {
+  type: InvocationErrorType.Timeout;
+};
+
+export type ResponseTooLargeInvocationError = BaseInvocationError & {
+  type: InvocationErrorType.ResponseTooLarge;
+};
+
+export type UnknownInvocationError = BaseInvocationError & {
+  type: InvocationErrorType.Unknown;
+};
+
+export type InvocationError =
+  | HttpStatusInvocationError
+  | RateLimitExceededInvocationError
+  | TimeoutInvocationError
+  | ResponseTooLargeInvocationError
+  | UnknownInvocationError;
+
+type MissingInvocationErrorTypes = Exclude<InvocationErrorType, InvocationError['type']>;
+/**
+ * We export this because unfortunately, TS doesn't let you ignore only an noUnusedLocals error without also
+ * suppressing the TS error we *want* to see.
+ * @hidden
+ */
+export type _ensureInvocationErrorEnumCompletion = Assert<MissingInvocationErrorTypes extends never ? true : false>;
```

Comment:
Say a bit more about what this is detecting? This is making sure the union type contains everything in our enum?